### ISSUE TITLE
Machine-readable output

### ISSUE THREAD

**Issue by fenhl** [rust-lang-nursery/rustup.rs#450]
I used to have a function in my shell that shows when the current directory is running on a multirust toolchain other than stable. The current formatting is making the port to rustup.rs difficult. An option to generate machine-readable output (preferably JSON) on relevant commands like `rustup show` and `rustup override list` would be great.

**Comment by brson** [comment]
Yes indeed. Sorry for breaking your stuff, but I did make the decision to cater to human-reading first.

Since this is mostly for shell scripts I'd guess it's best to do something simple and line-oriented. The default 'smart' behavior of `show` where it decides how much to show or not would be bad.

Perhaps a scheme like

* `rustup show active-toolchain --machine-readable`
* `rustup show active-toolchain-version --machine-readable`
* `rustup toolchain list --machine-readable`

**Comment by matklad** [comment]
>Since this is mostly for shell scripts

It would be useful in IntelliJ Rust as well, though I would also prefer simple text format.

**Comment by dherman** [comment]
This looks like it could be useful for Neon too -- +1 to shell script-friendliness. For the time being I may end up parsing the output of `rustc --version` or `rustup toolchain list`, but a stable machine-readable output would be awesome!

**Comment by alexreg** [comment]
Any progress on this?

**Comment by QAston** [comment]
Still needed

**Comment by camsteffen** [comment]
Found this because I wanted to do something like:

```shell
cd some/path
toolchain_name=$(rustup show active-toolchain)
cd some/other/path
RUSTUP_TOOLCHAIN=$toolchain_name cargo ...
```
